sdk/core/azure-core/inc/azure/core/logging/logging.hpp

```diff
@@ -9,143 +9,51 @@
 
 #pragma once
 
-#include "azure/core/dll_import_export.hpp"
-
 #include <functional>
-#include <initializer_list>
-#include <set>
 #include <string>
-#include <utility>
 
 namespace Azure { namespace Core { namespace Logging {
-  class LogClassification;
-  class LogClassifications;
+  /**
+   * @brief Log message level.
+   */
+  enum class LogLevel
```

Comment:
If these were flags then you could pick which type you want and it isn't stacking.
Error = 1,
Warning = 2, 
Informational = 4,
Verbose = 8,
...

Setting the log level to 0 would be == no logging

Reply:
If you want no logging, you set listener to `nullptr`.
I don't think we need flags - who would want verbose and warnings, but not informational and errors? I think, it should be "maximum level", and that's it. If you want no logging, you unset the logger, what's the point to have logger that will get no messages?

Reply:
You have to think about different concerns here. For example, it is common that an app will set one callback for the lifetime of the app but the level will change dynamically over the app's lifetime. To turn off logging, setting a null callback would work but then that code would have to know the name of the callback function to turn it back on again. Whereas the legal set of levels (error, warning, etc.) is always known globally throughout the program.

Reply:
I think of a LogLevel more as sound volume level, and not as bit flags.

Reply:
I agree

Reply:
But let's say if you really want Warnings and Informational but not errors and Verbose for some reason, you can filter in your callback.

Reply:
What if I want all logging, or as @RickWinter mentioned both error and warning?

The bit flag approach makes the most sense to achieve that. If I pick Warning, am I automatically getting both Warning and Error?

Reply:
Yes, this is how it works in most (maybe all) other languages. Logging is s scale; Verbose gives you everything.
Each log level is a severity and one severity implies the higher severities. It's not useful to ask for warnings and not ask for errors.